embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/comms/delivery/v2/SchedulingService.kt

```diff
@@ -0,0 +1,111 @@
+package io.embrace.android.embracesdk.internal.comms.delivery.v2
+
+import io.embrace.android.embracesdk.internal.clock.Clock
+import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
+import io.embrace.android.embracesdk.internal.worker.BackgroundWorker
+import io.embrace.android.embracesdk.internal.worker.ScheduledWorker
+import java.io.File
+import java.util.Queue
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.Future
+import java.util.concurrent.TimeUnit
+
+interface SchedulingService {
+    fun checkPendingPayloads()
+}
+
+internal class SchedulingServiceImpl(
+    private val cacheService: CacheService2,
+    private val executionService: ExecutionService,
+    private val schedulingWorker: ScheduledWorker,
+    private val worker: BackgroundWorker,
+    private val clock: Clock
+) : SchedulingService {
+
+    private val retryQueue: Queue<File> = ConcurrentLinkedQueue()
+
+    override fun checkPendingPayloads() {
+        schedulingWorker.submit {
+            var ready = readyToSend()
+            while (ready) {
+                runCatching {
+                    getNextReadyPayload()?.let { payload ->
+                        val response = send(payload).get()
+                        if (response is ApiResponse.TooManyRequests) {
+                            blockEndpoint(response)
+                        }
+                    }
+                }
+
+                ready = readyToSend()
```

Comment:
Is this necessary?

Reply:
The intent here is to check to see if the SDK and the execute environment is now ready to send requests. Thinking more about it, if we can just have the component react to any state changes that brings the SDK back on line, we don't need to check at each iteration. 

Let me take a look to see if I can remove.